src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -884,6 +887,55 @@ public async void StartNewReleaseAsync(string accountName, string projectName, A
                 body,
                 versionOverride: "5.0-preview.3",
                 baseAddressSubpath: "vsrm.");
+
+            return content.GetValue("id").ToObject<int>();
+        }
+
+        /// <summary>
+        /// Block until the release with ID informed finishes execution.
+        /// Checks the status of the release every 15 seconds.
+        /// </summary>
+        /// <param name="accountName"></param>
+        /// <param name="projectName"></param>
+        /// <param name="releaseId"></param>
+        public void WaitUntilCompleted(string accountName, string projectName, int releaseId)
+        {
+            while (true)
+            {
+                AzureDevOpsRelease rel = GetReleaseAsync(accountName, projectName, releaseId).GetAwaiter().GetResult();
+
+                if (rel.Environments[0].Status != "notStarted"
```

Comment:
I'd probably do a switch/case here to make it less noisy... I'd probably also ToLowerInvariant() the value, because these are... suspiciously cased.